### PR TITLE
Add option to set number of retries when no response

### PR DESCRIPTION
**Pull request by nolanmar511**
Related to addressing https://github.com/stephenplusplus/gcp-metadata/issues/10

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -158,6 +158,14 @@ describe('retry-request', function () {
       });
     });
 
+    it('should allow overriding noResponseRetries', function (done) {
+      var opts = { noResponseRetries: 0 };
+      
+      retryRequest(URI_NON_EXISTENT, opts, function () {
```
Comment (kjin) [inline test.js]: Can you add a check to see that the `'error'` event is emitted exactly once?

File: test.js [hunk 2]
```diff
@@ -120,7 +120,7 @@ describe('retry-request', function () {
         .on('error', done);
     });
   });
-
+
```
Comment (kjin) [inline test.js]: nit: whitespace

**Comment by nolanmar511** [comment]
PTAL

**Comment by ofrobots** [comment]
ping @stephenplusplus

**Comment by stephenplusplus** [comment]
Hey all, please take a look at my quick adjustments.

**Comment by ofrobots** [comment]
@stephenplusplus your changes lgtm, thanks!

**Comment by stephenplusplus** [comment]
Published 3.1.0

**Comment by stephenplusplus** [comment]
Thanks @nolanmar511!

**Pull request merged**